### PR TITLE
Update MSPX.md

### PR DESCRIPTION
**Pull request by b14ckyy**
Add INAV Setup steps

### CODE REVIEW

File: docs/MSPX.md [hunk 1]
```diff
@@ -19,5 +17,15 @@ Further parameter settings:
 - "Rx Snd RcChannel": When set to "rc override" or "rc channels", SET_RAW_RC MSP messages are sent to the flight controller. In the flight controller configure the receiver to the MSP protocol. This allows one to avoid the extra wire for CRSF.
 - "Rx Snd RadioStat": has no effect.
 
+## INAV Configuration
+
+To use a mLRS Receiver with INAV in MspX mode, the following settings have to be applied:
+- Enable MSP for the Serial Port the mLRS receiver is conencted to (do not enable Serial RX)
+- Set the Baud-Rate to 115200 or higher (Only INAV 7.1 and 8.0-dev are tested, Version 6.0 and newer should also work)
+- INAV 8.0 and later will also support Baud 230400 but with no noticeable difference in performance or stability 
+![image](https://github.com/user-attachments/assets/c359f819-66dd-453c-8883-f2498103c7a6)
```
Comment (jlpoltrack) [inline docs/MSPX.md:26]: Could you add a comment something like 'An example for UART2 is shown below' and/or maybe highlight the row with a box?

File: docs/MSPX.md [hunk 2]
```diff
@@ -19,5 +17,15 @@ Further parameter settings:
...
 
+- In the Receiver Tab, select the Receiver Mode Type to MSP and save settings
+![image](https://github.com/user-attachments/assets/75cb76b9-6487-4d64-a538-b46d4ca446bf)
```
Comment (jlpoltrack) [inline docs/MSPX.md:29]: Same thought as above - maybe highlight the section with a box to make it more obvious.

File: docs/MSPX.md [hunk 3]
```diff
@@ -19,5 +17,15 @@ Further parameter settings:
...
 
+You should now be able to detect channel values, if your radio is connected. No further settings are needed and telemetry will work for OpenTX/EdgeTX radios after Scanning for sensors.
```
Comment (jlpoltrack) [inline docs/MSPX.md:31]: Nitpick, order of phrases:

'If your radio is connected, you should now be able to see the channel values update.'

'scanning' lowercase s

**Comment by jlpoltrack** [comment]
Thanks - added a few small comments.  In additions where should this live on the main ReadMe?  Advanced Options?

**Comment by olliw42** [comment]
MANY THX. I second jlp's comments. In the pics I would not know where to look at :)

In the first list I think the last bullet could go first, and the first two bullets could become one 2nd bullet, the compression thing could be just a side note (it's really only one message which is compressed)(I would wish it would be more, but...)

> where should this live on the main ReadMe? Advanced Options?

I think that's a most important question, which also affects how we want to present this. Currently it's a bit like an odd feature "MspX". It could also be rather INAV/MSP, with mspX being just a comment.
It could go into Advanced Options, but I wonder if it might not be placed in Setups. If so it would "screw" up things a bit since most setups are for ArduPilot/PX4/MAVLink ... which is maybe not so clear then when INAV/MSP is added. So a bit of rewording/restructuring might be needed in this section then. I do have no specifc suggestions though.

just my 2 cents

**Comment by b14ckyy** [comment]
> Thanks - added a few small comments.  In additions where should this live on the main ReadMe?  Advanced Options?

Thanks I will address these in a bit.

**Comment by b14ckyy** [comment]
all changes made on the document and rephrased the INAV Version differences in a separate test block. Should be clearer now and more streamlined.

**Comment by olliw42** [comment]
I see your point.

Pl feel free to make suggestions and work this out with @jlpoltrack :)

I'm just adding my 2 cents every once in a while here :D

File: docs/MSPX.md [hunk 4]
```diff
@@ -19,5 +17,28 @@ Further parameter settings:
...
 - "Rx Snd RadioStat": has no effect.
 
+## INAV Configuration
+
+MspX is tested and verified with INAV 7.1 as well as 8.0 development builds but should also work with INAV 6.0 and newer. The baudrate should be set to 115200 as lower baud rates have shown irregular telemetry update rates on the radio due to the lack of flow control on the INAV side. 
+
+INAV 8.0 and later will also support a baud rate of 230400 but no change in performance or stability was noticed. Additionally with INAV 8.0 the performance of the serial link to a Ground Control Station (GCS) will be increased, due to some optimizations in INAV. 
+
+To use a mLRS Receiver with INAV in MspX mode, the following settings have to be applied:
+- Enable MSP for the Serial Port the mLRS receiver is conencted to (UART 2 is recommended on most STM32 Flight-Controllers; do not enable "Serial RX")
```
Comment (jlpoltrack) [inline docs/MSPX.md:27]: Spelling - 'conencted'

**Comment by jlpoltrack** [comment]
> A specific guide for things like Px4, Ardupilot and INAV setup would be nice. Even if it partially just has links to the necessary info docs in the right order.

In the short term - would you view this page as specific enough as a setup for INAV?  If so, I wonder if we rename the page as something like 'MspX for INAV'

**Comment by b14ckyy** [comment]
as INAV is the only system validated for use with MSP (when I find time I can see if it works with BF) it would make sense. 
the INAV guide is generally very short as there is really not much to set up compared to Mavlink. 

Alternatively I could extend the guide with a Mavlink Config as well and we split the mspX feature description from the INAV guide to make it a separate doc? So leave the file as it is and my new added part goes into a new INAV.md file?

**Comment by jlpoltrack** [comment]
> Alternatively I could extend the guide with a Mavlink Config as well and we split the mspX feature description from the INAV guide to make it a separate doc? So leave the file as it is and my new added part goes into a new INAV.md file?

Would a second page make sense here instead covering MAVLink on INAV or would you prefer all of the INAV stuff on one page?

**Comment by b14ckyy** [comment]
with the minimal settings needed (Its essentially just 2 parameters for MSP and 3 for MAVlink to be set) I am fine with all INAV stuff in one page. 
We can then link to the MspX, MavlinkX and CRSF info pages as well as the INAV Lua and Yapuu Lua as examples.

**Comment by jlpoltrack** [comment]
> More important question is: Would it be ok to have the INAV MAVlink config covered as this will only work with 8.0 and beyond. But that's still in development (although nothing will change until release in the setup process) or should we hold out on this till at least the RC phase?

With the single page approach - it would be good to have a section at the start outlining the Pros / Cons / Caveats of using MspX or MAVLink on INAV. Items like MAVLink support still in pre-release could be covered there.

**Comment by b14ckyy** [comment]
Yeah the Pro/Con of each also went through my head after my last post. 
I will update the PR after work and then you can tell me what you think.

**Comment by olliw42** [comment]
@b14ckyy @jlpoltrack any progress on this ?
just wondering

**Comment by b14ckyy** [comment]
Sorry had a bad week and was barely active anywhere.
Will finish it this weekend.

**Comment by olliw42** [comment]
oh, sorry to hear
no pressure, was just curious

**Comment by jlpoltrack** [comment]
Addressed by https://github.com/olliw42/mLRS-docu/pull/189

**Pull request closed without merging**